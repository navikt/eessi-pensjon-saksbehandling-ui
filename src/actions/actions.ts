export interface Action<T> {
  type: string,
  payload?: T
}

export interface JustTypeAction {
  type: string
}

export interface PayloadError {
  error: any
}
