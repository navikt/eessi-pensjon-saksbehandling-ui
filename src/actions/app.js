import * as types from '../constants/actionTypes'
import * as urls from '../constants/urls'
import * as api from './api'

export function login () {
  let currentHost = window.location.origin // http://hostname
  let redirect = currentHost
  let context = encodeURIComponent(window.location.pathname + window.location.search)

  let newUrl = urls.LOGIN_URL + '?redirect=' + redirect + '&context=' + context
  window.location.href = newUrl

  return {
    type: types.APP_LOGIN_REQUEST
  }
}

export function logout () {
  let redirectUrl = urls.LOGOUT_URL
  window.location.href = redirectUrl
  return {
    type: types.APP_LOGOUT_REQUEST
  }
}

export function getUserInfo () {
  return api.call({
    url: urls.API_USERINFO_URL,
    type: {
      request: types.APP_USERINFO_REQUEST,
      success: types.APP_USERINFO_SUCCESS,
      failure: types.APP_USERINFO_FAILURE
    }
  })
}

export function getAndPrefillPersonName () {
  return function (dispatch) {
    dispatch({ type: types.APP_GET_AND_PREFILL_PERSON_NAME })
    try {
      dispatch(getPersonData())
      dispatch(suggestPersonNameFromUsernameIfNotInState())
    } catch (error) {
      dispatch({ type: types.APP_GET_AND_PREFILL_PERSON_NAME_FAILURE, payload: error })
    }
  }
}

export function getPersonData () {
  return api.call({
    url: urls.API_PERSONDATA_URL,
    type: {
      request: types.APP_PERSONDATA_REQUEST,
      success: types.APP_PERSONDATA_SUCCESS,
      failure: types.APP_PERSONDATA_FAILURE
    }
  })
}

export function suggestPersonNameFromUsernameIfNotInState () {
  return function (dispatch, globalState) {
    if (!globalState.pinfo.person.nameAtBirth) {
      let lastName = globalState.app.lastName
      return dispatch({
        type: types.PINFO_PERSON_SET,
        payload: {
          nameAtBirth: lastName
        }
      })
    }
  }
}

export function clearData () {
  return {
    type: types.APP_CLEAR_DATA
  }
}

export function setReferrer (referrer) {
  return {
    type: types.APP_REFERRER_SET,
    payload: {
      referrer: referrer
    }
  }
}

export function registerDroppable (id, ref) {
  return {
    type: types.APP_DROPPABLE_REGISTER,
    payload: {
      id: id,
      ref: ref
    }
  }
}

export function unregisterDroppable (id) {
  return {
    type: types.APP_DROPPABLE_UNREGISTER,
    payload: {
      id: id
    }
  }
}
