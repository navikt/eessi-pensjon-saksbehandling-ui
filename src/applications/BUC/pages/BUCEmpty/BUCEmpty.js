import React, { useState } from 'react'
import PT from 'prop-types'

import { Nav } from 'eessi-pensjon-ui'
import MonitorPNG from 'resources/images/artwork/dataskjerm.png'
import CupPNG from 'resources/images/artwork/kop.png'
import MousePNG from 'resources/images/artwork/NAVmusematte.png'
import MapPNG from 'resources/images/artwork/saksstatus.png'
import BUCFooter from 'applications/BUC/components/BUCFooter/BUCFooter'

import './BUCEmpty.css'

const BUCEmpty = ({ actions, aktoerId, onBUCNew, rinaUrl, sakId, t }) => {
  const [_sakId, setSakId] = useState(sakId)
  const [_aktoerId, setAktoerId] = useState(aktoerId)
  const [validation, setValidation] = useState(undefined)

  const onAktoerIdChange = (e) => {
    setValidation(undefined)
    setAktoerId(e.target.value.trim())
  }

  const onSubmitAktoerId = () => {
    if (!_aktoerId || !_aktoerId.match(/^\d+$/)) {
      setValidation(t('buc:validation-noAktoerId'))
    } else {
      actions.setStatusParam('aktoerId', _aktoerId)
    }
  }

  const onSakIdChange = (e) => {
    setValidation(undefined)
    setSakId(e.target.value.trim())
  }

  const onSubmitSakId = () => {
    if (!_sakId || !_sakId.match(/^\d+$/)) {
      setValidation(t('buc:validation-noSakId'))
    } else {
      actions.setStatusParam('sakId', _sakId)
    }
  }

  return (
    <>
      <div className='panel a-buc-p-bucempty s-border d-flex'>
        <div className='a-buc-p-bucempty__artwork'>
          <img alt='' className='monitor' src={MonitorPNG} />
          <img alt='' className='cup' src={CupPNG} />
          <img alt='' className='mouse' src={MousePNG} />
          <img alt='' className='map' src={MapPNG} />
        </div>
        <Nav.Undertittel className='a-buc-bucempty__title mb-3'>
          <Nav.Lenke
            id='a-buc-p-bucempty__newbuc-link-id'
            className='a-buc-p-bucempty__newbuc-link'
            href='#' onClick={onBUCNew}
          >
            {t('buc:form-empty-startANewCase')}
          </Nav.Lenke>
        </Nav.Undertittel>
        {!aktoerId ? (
          <div className='a-buc-p-bucempty__form'>
            <Nav.Input
              id='a-buc-p-bucempty__aktoerid-input-id'
              className='a-buc-p-bucempty__aktoerid-input'
              label={t('ui:aktoerId')}
              value={_aktoerId || ''}
              bredde='fullbredde'
              onChange={onAktoerIdChange}
              feil={validation ? { feilmelding: validation } : null}
            />
            <Nav.Hovedknapp
              id='a-buc-p-bucempty__aktoerid-button-id'
              className='a-buc-p-bucempty__aktoerid-button ml-3'
              onClick={onSubmitAktoerId}
            >
              {t('ui:add')}
            </Nav.Hovedknapp>
          </div>
        ) : null}
        {!sakId ? (
          <div className='a-buc-p-bucempty__form'>
            <Nav.Input
              id='a-buc-p-bucempty__sakid-input-id'
              className='a-buc-p-bucempty__sakid-input'
              label={t('buc:form-caseId')}
              value={_sakId || ''}
              bredde='fullbredde'
              onChange={onSakIdChange}
              feil={validation ? { feilmelding: validation } : null}
            />
            <Nav.Hovedknapp
              id='a-buc-p-bucempty__sakid-button-id'
              className='a-buc-p-bucempty__sakid-button ml-3'
              onClick={onSubmitSakId}
            >
              {t('ui:add')}
            </Nav.Hovedknapp>
          </div>
        ) : null}
      </div>
      <BUCFooter className='w-100 mt-2 mb-2' rinaUrl={rinaUrl} t={t} />
    </>
  )
}

BUCEmpty.propTypes = {
  actions: PT.object.isRequired,
  aktoerId: PT.string,
  onBUCNew: PT.func.isRequired,
  sakId: PT.string,
  t: PT.func.isRequired
}

export default BUCEmpty
