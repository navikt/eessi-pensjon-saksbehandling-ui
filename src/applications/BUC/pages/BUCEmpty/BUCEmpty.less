@import '../../../../constants';

.a-buc-p-bucempty {

  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;

  &__artwork {
    position: relative;
    height: 300px;
    width: 250px;

    .monitor {
      width: 250px;
      height: 200px;
    }

    .cup {
      width: 33px;
      height: 30px;
      top: 170px;
      left: 255px;
    }

    .mouse {
      width: 60px;
      height: 80px;
      left: 190px;
      top: 180px;
    }

    .map {
      width: 70px;
      height: 100px;
      top: 20px;
      left: 90px;
    }

    img {
      position: absolute;
    }
  }

  &__form {
    display: flex;
    flex-direction: row;
    align-items: flex-end;

    .skjemaelement {
      margin-bottom: 0px !important
    }
  }
}

.highContrast .a-buc-p-bucempty {
  background: @black;
}
