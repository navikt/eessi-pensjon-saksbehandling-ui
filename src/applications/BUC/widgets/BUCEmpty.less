.a-buc-bucempty {

  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;

  &__artwork {
    position: relative;
    height: 300px;
    width: 250px;

    .monitor {
      width: 250px;
      height: 200px;
    }

    .cup {
      width: 33px;
      height: 30px;
      top: 170px;
      left: 255px;
    }

    .mouse {
      width: 60px;
      height: 80px;
      left: 190px;
      top: 180px;
    }

    .map {
      width: 70px;
      height: 100px;
      top: 20px;
      left: 90px;
    }

    img {
      position: absolute;
    }
  }

  &__aktoerid-div {
    display: flex;
    flex-direction: column;
    align-items: flex-start;

    .skjemaelement {
      margin-bottom: 0px !important
    }
  }

  &__aktoerid-button {
    margin-top: 0.75rem;
  }
}
