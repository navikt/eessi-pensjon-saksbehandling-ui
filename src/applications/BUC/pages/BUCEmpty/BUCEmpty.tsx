import { setStatusParam } from 'actions/app'
import BUCFooter from 'applications/BUC/components/BUCFooter/BUCFooter'
import MonitorPNG from 'assets/images/artwork/dataskjerm.png'
import CupPNG from 'assets/images/artwork/kop.png'
import MousePNG from 'assets/images/artwork/NAVmusematte.png'
import MapPNG from 'assets/images/artwork/saksstatus.png'
import { HorizontalSeparatorDiv, VerticalSeparatorDiv } from 'components/StyledComponents'
import { State } from 'declarations/reducers'
import { RinaUrl } from 'declarations/types'
import { standardLogger } from 'metrics/loggers'
import { Hovedknapp } from 'nav-frontend-knapper'
import Lenke from 'nav-frontend-lenker'
import { Input } from 'nav-frontend-skjema'
import { Undertittel } from 'nav-frontend-typografi'
import { theme, themeHighContrast } from 'nav-styled-component-theme'
import PT from 'prop-types'
import React, { useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { useDispatch, useSelector } from 'react-redux'
import styled, { ThemeProvider } from 'styled-components'

export interface BUCEmptySelector {
  rinaUrl: RinaUrl | undefined
  highContrast: boolean
}

const mapState = (state: State): BUCEmptySelector => ({
  highContrast: state.ui.highContrast,
  rinaUrl: state.buc.rinaUrl
})

export interface BUCEmptyProps {
  aktoerId?: string;
  onBUCNew: () => void;
  sakId?: string;
}

const BUCEmptyDiv = styled.div`
  display: flex;
  border: 1px solid ${({ theme }: any) => theme['main-font-color']};
  justify-content: center;
  align-items: center;
  flex-direction: column;
  padding: 2rem;
  background-color: ${({ theme }: any) => theme['main-background-color']};
`
const BUCEmptyArtwork = styled.div`
  position: relative;
  height: 300px;
  width: 250px;

  .monitor {
    width: 250px;
    height: 200px;
  }

  .cup {
    width: 33px;
    height: 30px;
    top: 170px;
    left: 255px;
  }

  .mouse {
    width: 60px;
    height: 80px;
    left: 190px;
    top: 180px;
  }

  .map {
    width: 70px;
    height: 100px;
    top: 20px;
    left: 90px;
  }

  img {
    position: absolute;
  }
`
const BUCEmptyForm = styled.div`
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  .skjemaelement {
    margin-bottom: 0px !important
  }
`

const BUCEmpty: React.FC<BUCEmptyProps> = ({
  aktoerId, onBUCNew, sakId
}: BUCEmptyProps): JSX.Element => {
  const [_sakId, setSakId] = useState<string | undefined>(sakId)
  const [_aktoerId, setAktoerId] = useState<string | undefined>(aktoerId)
  const [validation, setValidation] = useState<string | undefined>(undefined)
  const dispatch = useDispatch()
  const { highContrast, rinaUrl }: BUCEmptySelector = useSelector<State, BUCEmptySelector>(mapState)
  const { t } = useTranslation()

  useEffect(() => {
    standardLogger('buc.empty.entrance')
  }, [])

  const onAktoerIdChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    setValidation(undefined)
    setAktoerId(e.target.value.trim())
  }

  const onSubmitAktoerId = (): void => {
    if (!_aktoerId || !_aktoerId.match(/^\d+$/)) {
      setValidation(t('buc:validation-noAktoerId'))
    } else {
      dispatch(setStatusParam('aktoerId', _aktoerId))
    }
  }

  const onSakIdChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    setValidation(undefined)
    setSakId(e.target.value.trim())
  }

  const onSubmitSakId = (): void => {
    if (!_sakId || !_sakId.match(/^\d+$/)) {
      setValidation(t('buc:validation-noSakId'))
    } else {
      dispatch(setStatusParam('sakId', _sakId))
    }
  }

  return (
    <ThemeProvider theme={highContrast ? themeHighContrast : theme}>
      <BUCEmptyDiv>
        <BUCEmptyArtwork>
          <img alt='' className='monitor' src={MonitorPNG} />
          <img alt='' className='cup' src={CupPNG} />
          <img alt='' className='mouse' src={MousePNG} />
          <img alt='' className='map' src={MapPNG} />
        </BUCEmptyArtwork>
        <VerticalSeparatorDiv />
        <Undertittel>
          <Lenke
            data-tedtIid='a-buc-p-bucempty__newbuc-link-id'
            href='#'
            onClick={onBUCNew}
          >
            {t('buc:form-empty-startANewCase')}
          </Lenke>
        </Undertittel>
        <VerticalSeparatorDiv />
        {!aktoerId && (
          <BUCEmptyForm>
            <Input
              data-testId='a-buc-p-bucempty__aktoerid-input-id'
              label={t('ui:aktoerId')}
              value={_aktoerId || ''}
              bredde='fullbredde'
              onChange={onAktoerIdChange}
              feil={validation || false}
            />
            <HorizontalSeparatorDiv />
            <Hovedknapp
              data-testId='a-buc-p-bucempty__aktoerid-button-id'
              onClick={onSubmitAktoerId}
            >
              {t('ui:add')}
            </Hovedknapp>
          </BUCEmptyForm>
        )}
        {!sakId && (
          <BUCEmptyForm>
            <Input
              data-testId='a-buc-p-bucempty__sakid-input-id'
              label={t('buc:form-caseId')}
              value={_sakId || ''}
              bredde='fullbredde'
              onChange={onSakIdChange}
              feil={validation || false}
            />
            <HorizontalSeparatorDiv />
            <Hovedknapp
              data-testId='a-buc-p-bucempty__sakid-button-id'
              onClick={onSubmitSakId}
            >
              {t('ui:add')}
            </Hovedknapp>
          </BUCEmptyForm>
        )}
      </BUCEmptyDiv>
      {rinaUrl && (<BUCFooter />)}
    </ThemeProvider>
  )
}

BUCEmpty.propTypes = {
  aktoerId: PT.string,
  onBUCNew: PT.func.isRequired,
  sakId: PT.string
}

export default BUCEmpty
