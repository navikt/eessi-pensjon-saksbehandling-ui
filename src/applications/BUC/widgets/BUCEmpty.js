import React, { useState } from 'react'
import PT from 'prop-types'

import { Flatknapp, Hovedknapp, Input, Lenke, Normaltekst, ToggleKnapp, Undertittel } from 'components/ui/Nav'

import * as MonitorPNG from 'resources/images/artwork/dataskjerm.png'
import * as CupPNG from 'resources/images/artwork/kop.png'
import * as MousePNG from 'resources/images/artwork/NAVmusematte.png'
import * as MapPNG from 'resources/images/artwork/saksstatus.png'

import './BUCEmpty.css'

const BUCEmpty = (props) => {
  const { actions, aktoerId, bucs, getBucs, gettingBUCs, onBUCNew, sakId, t } = props
  const [ _sakId, setSakId ] = useState(sakId)
  const [ _aktoerId, setAktoerId ] = useState(aktoerId)
  const [ validation, setValidation ] = useState(undefined)

  const onAktoerIdChange = (e) => {
    setValidation(undefined)
    setAktoerId(e.target.value.trim())
  }

  const onSubmitAktoerId = (e) => {
    if (!_aktoerId || !_aktoerId.match(/^\d+$/)) {
      setValidation(t('buc:validation-noAktoerId'))
    } else {
      actions.setStatusParam('aktoerId', _aktoerId)
    }
  }

  const onSakIdChange = (e) => {
    setValidation(undefined)
    setSakId(e.target.value.trim())
  }

  const onSubmitSakId = (e) => {
    if (!_sakId || !_sakId.match(/^\d+$/)) {
      setValidation(t('buc:validation-noSakId'))
    } else {
      actions.setStatusParam('sakId', _sakId)
    }
  }

  return <div className='a-buc-bucempty'>
    <div className='a-buc-bucempty__artwork'>
      <img alt='' className='monitor' src={MonitorPNG} />
      <img alt='' className='cup' src={CupPNG} />
      <img alt='' className='mouse' src={MousePNG} />
      <img alt='' className='map' src={MapPNG} />
    </div>
    <Undertittel className='mb-3'>
      <Lenke href='#' onClick={onBUCNew}>
        {t('buc:form-empty-startANewCase')}
      </Lenke>
    </Undertittel>
    {!aktoerId ? <div className='a-buc-bucempty__aktoerid-div'>
      <Input
        className='a-buc-bucempty__aktoerid-input'
        id='a-buc-bucempty__aktoerid-input-id'
        label={t('ui:aktoerId')}
        value={_aktoerId || ''}
        bredde='fullbredde'
        onChange={onAktoerIdChange}
        feil={validation ? { feilmelding: validation } : null} />
      <Hovedknapp
        id='a-buc-bucempty__aktoerid-button-id'
        className='a-buc-bucempty__aktoerid-button ml-3'
        onClick={onSubmitAktoerId}>
        {t('ui:add')}
      </Hovedknapp>
    </div> : null}
    {!sakId ? <div className='a-buc-bucempty__sakid-div'>
      <Input
        className='a-buc-bucempty__sakid-input'
        id='a-buc-bucempty__sakid-input-id'
        label={t('ui:caseId')}
        value={_sakId || ''}
        bredde='fullbredde'
        onChange={onSakIdChange}
        feil={validation ? { feilmelding: validation } : null} />
      <Hovedknapp
        id='a-buc-bucempty__sakid-button-id'
        className='a-buc-bucempty__sakid-button ml-3'
        onClick={onSubmitSakId}>
        {t('ui:add')}
      </Hovedknapp>
    </div> : null}
  </div>
}

BUCEmpty.propTypes = {
  actions: PT.object.isRequired,
  aktoerId: PT.string,
  bucs: PT.array,
  getBucs: PT.func.isRequired,
  gettingBUCs: PT.bool.isRequired,
  onBUCNew: PT.func.isRequired,
  sakId: PT.string,
  t: PT.func.isRequired
}

export default BUCEmpty
