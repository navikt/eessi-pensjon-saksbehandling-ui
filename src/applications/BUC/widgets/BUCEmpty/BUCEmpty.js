import React, { useState } from 'react'
import PT from 'prop-types'

import { Nav } from 'eessi-pensjon-ui'
import MonitorPNG from 'resources/images/artwork/dataskjerm.png'
import CupPNG from 'resources/images/artwork/kop.png'
import MousePNG from 'resources/images/artwork/NAVmusematte.png'
import MapPNG from 'resources/images/artwork/saksstatus.png'

import './BUCEmpty.css'

const BUCEmpty = ({ actions, aktoerId, onBUCNew, sakId, t }) => {
  const [_sakId, setSakId] = useState(sakId)
  const [_aktoerId, setAktoerId] = useState(aktoerId)
  const [validation, setValidation] = useState(undefined)

  const onAktoerIdChange = (e) => {
    setValidation(undefined)
    setAktoerId(e.target.value.trim())
  }

  const onSubmitAktoerId = () => {
    if (!_aktoerId || !_aktoerId.match(/^\d+$/)) {
      setValidation(t('buc:validation-noAktoerId'))
    } else {
      actions.setStatusParam('aktoerId', _aktoerId)
    }
  }

  const onSakIdChange = (e) => {
    setValidation(undefined)
    setSakId(e.target.value.trim())
  }

  const onSubmitSakId = () => {
    if (!_sakId || !_sakId.match(/^\d+$/)) {
      setValidation(t('buc:validation-noSakId'))
    } else {
      actions.setStatusParam('sakId', _sakId)
    }
  }

  return (
    <div className='panel a-buc-bucempty s-border d-flex'>
      <div className='a-buc-bucempty__artwork'>
        <img alt='' className='monitor' src={MonitorPNG} />
        <img alt='' className='cup' src={CupPNG} />
        <img alt='' className='mouse' src={MousePNG} />
        <img alt='' className='map' src={MapPNG} />
      </div>
      <Nav.Undertittel className='a-buc-bucempty__title mb-3'>
        <Nav.Lenke
          id='a-buc-bucempty__newbuc-link-id'
          className='a-buc-bucempty__newbuc-link'
          href='#' onClick={onBUCNew}
        >
          {t('buc:form-empty-startANewCase')}
        </Nav.Lenke>
      </Nav.Undertittel>
      {!aktoerId ? (
        <div className='a-buc-bucempty__form'>
          <Nav.Input
            id='a-buc-bucempty__aktoerid-input-id'
            className='a-buc-bucempty__aktoerid-input'
            label={t('ui:aktoerId')}
            value={_aktoerId || ''}
            bredde='fullbredde'
            onChange={onAktoerIdChange}
            feil={validation ? { feilmelding: validation } : null}
          />
          <Nav.Hovedknapp
            id='a-buc-bucempty__aktoerid-button-id'
            className='a-buc-bucempty__aktoerid-button ml-3'
            onClick={onSubmitAktoerId}
          >
            {t('ui:add')}
          </Nav.Hovedknapp>
        </div>
      ) : null}
      {!sakId ? (
        <div className='a-buc-bucempty__form'>
          <Nav.Input
            id='a-buc-bucempty__sakid-input-id'
            className='a-buc-bucempty__sakid-input'
            label={t('buc:form-caseId')}
            value={_sakId || ''}
            bredde='fullbredde'
            onChange={onSakIdChange}
            feil={validation ? { feilmelding: validation } : null}
          />
          <Nav.Hovedknapp
            id='a-buc-bucempty__sakid-button-id'
            className='a-buc-bucempty__sakid-button ml-3'
            onClick={onSubmitSakId}
          >
            {t('ui:add')}
          </Nav.Hovedknapp>
        </div>
      ) : null}
    </div>
  )
}

BUCEmpty.propTypes = {
  actions: PT.object.isRequired,
  aktoerId: PT.string,
  onBUCNew: PT.func.isRequired,
  sakId: PT.string,
  t: PT.func.isRequired
}

export default BUCEmpty
