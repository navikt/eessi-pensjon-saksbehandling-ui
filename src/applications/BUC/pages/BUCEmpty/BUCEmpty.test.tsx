import { setStatusParam } from 'actions/app'
import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import { stageSelector } from 'setupTests'
import BUCEmpty, { BUCEmptyProps } from './BUCEmpty'

const defaultSelector = {
  rinaUrl: 'http://mock.url'
}

jest.mock('actions/app', () => ({
  setStatusParam: jest.fn()
}))

describe('applications/BUC/widgets/BUCEmpty/BUCEmpty', () => {
  let wrapper: ReactWrapper
  const initialMockProps: BUCEmptyProps = {
    onBUCNew: jest.fn()
  }

  beforeAll(() => {
    stageSelector(defaultSelector, {})
  })

  beforeEach(() => {
    wrapper = mount(<BUCEmpty {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Has proper HTML structure with forms when no aktoerId and sakId', () => {
    expect(wrapper.exists('.a-buc-p-bucempty')).toBeTruthy()
    expect(wrapper.exists('.a-buc-p-bucempty__artwork')).toBeTruthy()
    expect(wrapper.exists('.a-buc-p-bucempty__title')).toBeTruthy()
    expect(wrapper.exists('#a-buc-p-bucempty__newbuc-link-id')).toBeTruthy()
    expect(wrapper.exists('.a-buc-p-bucempty__form')).toBeTruthy()
    expect(wrapper.exists('#a-buc-p-bucempty__aktoerid-input-id')).toBeTruthy()
    expect(wrapper.exists('#a-buc-p-bucempty__aktoerid-button-id')).toBeTruthy()
    expect(wrapper.exists('#a-buc-p-bucempty__sakid-input-id')).toBeTruthy()
    expect(wrapper.exists('#a-buc-p-bucempty__sakid-button-id')).toBeTruthy()
  })

  it('Has proper HTML structure without forms when aktoerId and sakId', () => {
    const mockProps = {
      ...initialMockProps,
      aktoerId: '123',
      sakId: '456'
    }
    wrapper = mount(<BUCEmpty {...mockProps} />)
    expect(wrapper.exists('#a-buc-p-bucempty__aktoerid-input-id')).toBeFalsy()
    expect(wrapper.exists('#a-buc-p-bucempty__aktoerid-button-id')).toBeFalsy()
    expect(wrapper.exists('#a-buc-p-bucempty__sakid-input-id')).toBeFalsy()
    expect(wrapper.exists('#a-buc-p-bucempty__sakid-button-id')).toBeFalsy()
  })

  it('Goes to bucnew when button pressed', () => {
    wrapper.find('#a-buc-p-bucempty__newbuc-link-id').hostNodes().simulate('click')
    expect(initialMockProps.onBUCNew).toHaveBeenCalled()
  })

  it('Handles added aktoerId and sakId', () => {
    wrapper.find('#a-buc-p-bucempty__aktoerid-input-id').hostNodes().simulate('change', { target: { value: 'notvalid' } })
    wrapper.find('#a-buc-p-bucempty__aktoerid-button-id').hostNodes().simulate('click')
    wrapper.update()
    expect(wrapper.find('#a-buc-p-bucempty__aktoerid-input-id .skjemaelement__feilmelding').render().text()).toEqual('buc:validation-noAktoerId')

    wrapper.find('#a-buc-p-bucempty__aktoerid-input-id').hostNodes().simulate('change', { target: { value: '123' } })
    wrapper.find('#a-buc-p-bucempty__aktoerid-button-id').hostNodes().simulate('click')
    wrapper.update()
    expect(setStatusParam).toBeCalledWith('aktoerId', '123')

    wrapper.find('#a-buc-p-bucempty__sakid-input-id').hostNodes().simulate('change', { target: { value: 'notvalid' } })
    wrapper.find('#a-buc-p-bucempty__sakid-button-id').hostNodes().simulate('click')
    wrapper.update()
    expect(wrapper.find('#a-buc-p-bucempty__sakid-input-id .skjemaelement__feilmelding').render().text()).toEqual('buc:validation-noSakId')

    wrapper.find('#a-buc-p-bucempty__sakid-input-id').hostNodes().simulate('change', { target: { value: '123' } })
    wrapper.find('#a-buc-p-bucempty__sakid-button-id').hostNodes().simulate('click')
    wrapper.update()
    expect(setStatusParam).toBeCalledWith('sakId', '123')
  })
})
