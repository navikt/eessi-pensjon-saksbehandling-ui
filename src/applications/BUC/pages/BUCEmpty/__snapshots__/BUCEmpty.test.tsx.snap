// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications/BUC/components/BUCEmpty/BUCEmpty Render: match snapshot 1`] = `
.c9 {
  display: inline-block;
  margin-left: 1rem;
}

ui-fss.g18[id="c8"] {
  content: "c9,heHwTk,";
}

.c5 {
  margin-bottom: 1rem;
}

ui-fss.g19[id="c4"] {
  content: "c5,";
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-color: var(--a-border);
  border-style: solid;
  border-width: 1px;
  border-radius: 4px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 2rem;
  background-color: var(--a-component-background-alternate);
}

ui-fss.g42[id="c0"] {
  content: "c1,";
}

.c3 {
  position: relative;
  height: 300px;
  width: 250px;
}

.c3 .monitor {
  width: 250px;
  height: 200px;
}

.c3 .cup {
  width: 33px;
  height: 30px;
  top: 170px;
  left: 255px;
}

.c3 .mouse {
  width: 60px;
  height: 80px;
  left: 190px;
  top: 180px;
}

.c3 .map {
  width: 70px;
  height: 100px;
  top: 20px;
  left: 90px;
}

.c3 img {
  position: absolute;
}

ui-fss.g43[id="c2"] {
  content: "c3,";
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c7 .skjemaelement {
  margin-bottom: 0px !important;
}

ui-fss.g44[id="c6"] {
  content: "c7,";
}

<div
  class="c0 c1"
>
  <div
    class="c2 c3"
  >
    <img
      alt=""
      class="monitor"
      src="dataskjerm.png"
    />
    <img
      alt=""
      class="cup"
      src="kop.png"
    />
    <img
      alt=""
      class="mouse"
      src="NAVmusematte.png"
    />
    <img
      alt=""
      class="map"
      src="saksstatus.png"
    />
  </div>
  <div
    class="c4 c5"
  />
  <div
    class="c6 c7"
  >
    <div
      class="navds-form-field navds-form-field--medium"
    >
      <label
        class="navds-form-field__label navds-label"
        for="a-buc-p-bucempty--aktoerid-input-id"
      >
        ui:aktoerId
      </label>
      <input
        aria-invalid="false"
        class="navds-text-field__input navds-body-short navds-body-medium"
        data-testid="a-buc-p-bucempty--aktoerid-input-id"
        id="a-buc-p-bucempty--aktoerid-input-id"
        type="text"
        value=""
      />
      <div
        aria-live="polite"
        aria-relevant="additions removals"
        class="navds-form-field__error"
        id="textField-error-:r3:"
      />
    </div>
    <div
      class="c8 c9"
    />
    <button
      class="navds-button navds-button--primary navds-button--medium"
      data-testid="a-buc-p-bucempty--aktoerid-button-id"
    >
      <span
        aria-live="polite"
        class="navds-label"
      >
        ui:add
      </span>
    </button>
  </div>
  <div
    class="c6 c7"
  >
    <div
      class="navds-form-field navds-form-field--medium"
    >
      <label
        class="navds-form-field__label navds-label"
        for="a-buc-p-bucempty--sakid-input-id"
      >
        buc:form-caseId
      </label>
      <input
        aria-invalid="false"
        class="navds-text-field__input navds-body-short navds-body-medium"
        data-testid="a-buc-p-bucempty--sakid-input-id"
        id="a-buc-p-bucempty--sakid-input-id"
        type="text"
        value=""
      />
      <div
        aria-live="polite"
        aria-relevant="additions removals"
        class="navds-form-field__error"
        id="textField-error-:r4:"
      />
    </div>
    <div
      class="c8 c9"
    />
    <button
      class="navds-button navds-button--primary navds-button--medium"
      data-testid="a-buc-p-bucempty--sakid-button-id"
    >
      <span
        aria-live="polite"
        class="navds-label"
      >
        ui:add
      </span>
    </button>
  </div>
</div>
`;
